docs/contributor/arch/008-telemetry-flow-healthiness-status-api.md

```diff
@@ -0,0 +1,102 @@
+Ultimately, the user's primary concern is understanding whether the telemetry flow is functioning correctly. In case of any issues, the user has the following actionable steps:
+
+* Troubleshoot the backend.
+* Investigate backend connectivity.
+* Reduce ingestion.
+* Manually scale out the gateway (as long as no autoscaling capability is provided).
+
+In this scenario, we can assign the reason field a value that holds utmost significance for the user:
+```
+FullDataLoss > PartialDataLoss > HighBufferUtilization > GatewayThrottling > Healthy
+```
+
+The reasons will then be based on the following alert rules (an example for Metric Pipelines the actual PromQL expressions have to be defined later):
+| Alert Rule | Expression |
+| --- | --- |
+| GatewayExporterSentMetrics    | `sum(rate(otelcol_exporter_sent_metric_points{...}[5m])) > 0`           |
+| GatewayExporterDroppedMetrics  | `sum(rate(otelcol_exporter_send_failed_metric_points{...}[5m])) > 0`    |
+| GatewayReceiverRefusedMetrics  | `sum(rate(otelcol_receiver_refused_metric_points{...}[5m])) > 0`        |
+| GatewayExporterEnqueueFailed   | `sum(rate(otelcol_exporter_enqueue_failed_metric_points{...}[5m])) > 0` |
+| GatewayExporterQueueAlmostFull | `otelcol_exporter_queue_size / otelcol_exporter_queue_capacity > 0.8`   |                               |
+
+We could then map the alert rules to the reasons as follows:
+
+| Reason | Alert Rules |
+| --- | --- |
+| FullDataLoss           | **not** GatewayExporterSentMetrics **and** (GatewayExporterDroppedMetrics **or** GatewayExporterEnqueueFailed) |
+| PartialDataLoss        | GatewayExporterSentMetrics **and** (GatewayExporterDroppedMetrics **or** GatewayExporterEnqueueFailed)       |
+| HighBufferUtilization  | GatewayExporterQueueAlmostFull                                                                           |
+| GatewayThrottling      | GatewayReceiverRefusedMetrics                                                                            |
+| Healthy                | **not** (GatewayExporterDroppedMetrics **or** GatewayExporterEnqueueFailed **or** HighBufferUtilization **or** GatewayReceiverRefusedMetrics) |
+
+### Gateway/Agent Communication
+
+The mentioned events are about the Gateway Collector. Some issues might happen with the Agent, like:
+* Gateway can't be reached (network problem).
+* Gateway won't accept data (gateway throttling).
+
+In both cases, the Agent might drop data after too many retries. But since users usually can't do much about these issues, we won't include them in the TelemetryFlowHealthy condition.
+If there's throttling, scaling horizontally might help, but it's not much different from dealing with a high OTLP ingestion rate.
+
+In the future, we might think about adding a TelemetryCollectionHealthy condition for the Agent. This would help track issues like scraping problems by Prometheus Receiver.
```

Comment:
```suggestion
In the future, we might think about adding a `TelemetryCollectionHealthy` condition for the Agent. This would help track issues like scraping problems by Prometheus Receiver.
```